### PR TITLE
Antd updates

### PR DESCRIPTION
**Pull request by ChristopherChudzicki**
- Links to documentation: https://ant.design/docs/react/introduce
- Link to GitHub or NPM: https://github.com/ant-design/ant-design
- Type of contribution: addition

Other notes: This PR adds a few missing `antd` components, as well as props for some of them.

### CODE REVIEW

File: definitions/npm/antd_v3.x.x/flow_v0.25.x-/antd_v3.x.x.js [hunk 1]
```diff
@@ -27,7 +27,25 @@ declare module "antd" {
 
   declare export class Badge extends React$Component<{}> {}
 
-  declare export class Button extends React$Component<{}> {}
+  declare export type ButtonProps = {
+    disabled?: boolean,
+    ghost?: boolean,
+    href?: boolean,
```
Comment (pascalduez) [inline definitions/npm/antd_v3.x.x/flow_v0.25.x-/antd_v3.x.x.js]: `string`

File: definitions/npm/antd_v3.x.x/flow_v0.25.x-/antd_v3.x.x.js [hunk 2]
```diff
@@ -27,7 +27,25 @@ declare module "antd" {
...
+    htmlType?: string,
+    icon?: string,
+    loading?: boolean | { delay: number },
+    shape?: 'circle' | 'round',
+    size?: 'small' | 'large',
+    type?: 'primary' | 'ghost' | 'dashed' | 'danger' | 'default',
+    onClick?: (event?: SyntheticEvent<HTMLButtonElement>) => void,
+    block?: boolean
+  }
```
Comment (pascalduez) [inline definitions/npm/antd_v3.x.x/flow_v0.25.x-/antd_v3.x.x.js]: Missing `target?: string`

File: definitions/npm/antd_v3.x.x/flow_v0.25.x-/test_antd_v3.x.x.js [hunk 3]
```diff
@@ -442,6 +477,13 @@ describe("Tooltip", () => {
   it("is a react component", () => {
     const tooltip = <Tooltip title="hello" />;
   });
+  it("should accept only certain strings for placement prop", () => {
+    // Testing placement prop in particular just to verify sharing between
+    // Popconfirm, Popover, and Tooltip is working.
+    const good = <Popover placement='top' />;
```
Comment (pascalduez) [inline definitions/npm/antd_v3.x.x/flow_v0.25.x-/test_antd_v3.x.x.js]: copy pasta, must be `Tooltip`

File: definitions/npm/antd_v3.x.x/flow_v0.25.x-/antd_v3.x.x.js [hunk 4]
```diff
@@ -345,7 +363,41 @@ declare module "antd" {
...
+    visible?: boolean,
+    onVisibleChange?: (visible?: boolean) => void,
+    align?: Object
+  }
+
+  declare export type PopconfirmProps = {
+    cancelText?: string,
+    okText?: string,
+    okType?: 'primary' | 'dashed' | 'ghost' | 'danger' | 'default',
+    title?: 'string' | React$Node,
```
Comment (pascalduez) [inline definitions/npm/antd_v3.x.x/flow_v0.25.x-/antd_v3.x.x.js]: `React$Node` already allows strings

File: definitions/npm/antd_v3.x.x/flow_v0.25.x-/antd_v3.x.x.js [hunk 5]
```diff
@@ -345,7 +363,41 @@ declare module "antd" {
...
+    onCancel?: (event?: SyntheticEvent<HTMLElement>) => void,
+    onConfirm?: (event?: SyntheticEvent<HTMLElement>) => void,
+    icon?: React$Node
+  } & TooltipSharedProps
+
+  declare export class Popconfirm extends React$Component<PopconfirmProps> {}
+
+  declare export type PopoverProps = {
+    content?: 'string' | React$Node,
+    title?: 'string' | React$Node,
```
Comment (pascalduez) [inline definitions/npm/antd_v3.x.x/flow_v0.25.x-/antd_v3.x.x.js]: `React$Node` already allows strings

File: definitions/npm/antd_v3.x.x/flow_v0.25.x-/antd_v3.x.x.js [hunk 6]
```diff
@@ -345,7 +363,41 @@ declare module "antd" {
...
+  // These props are shared by Tooltip, Popconfirm, and Poopover
+  declare type TooltipSharedProps = {
+    arrowPointAtCenter?: boolean,
+    autoAdjustOverflow?: boolean,
+    defaultVisible?: boolean,
+    getPopupContainer?: (element?: HTMLElement) => HTMLElement,
+    mouseEnterDelay?: number,
+    mouseLeaveDelay?: number,
+    overlayClassName?: string,
+    overlayStyle?: Object,
```
Comment (pascalduez) [inline definitions/npm/antd_v3.x.x/flow_v0.25.x-/antd_v3.x.x.js]: `Object` === `any` 

`CSSStyleDeclaration` or `{ +[string]: mixed }`

File: definitions/npm/antd_v3.x.x/flow_v0.25.x-/antd_v3.x.x.js [hunk 7]
```diff
@@ -345,7 +363,41 @@ declare module "antd" {
...
+    placement?: 'top' | 'left' | 'right' | 'bottom' | 'topLeft' | 'topRight' | 'bottomLeft' | 'bottomRight' | 'leftTop' | 'leftBottom' | 'rightTop' | 'rightBottom',
+    trigger?: 'hover' | 'focus' | 'click' | 'contextMenu',
+    visible?: boolean,
+    onVisibleChange?: (visible?: boolean) => void,
+    align?: Object
```
Comment (pascalduez) [inline definitions/npm/antd_v3.x.x/flow_v0.25.x-/antd_v3.x.x.js]: Same, `Object` === `any`

**Comment by ChristopherChudzicki** [comment]
@pascalduez Thanks for the review, and for finding those errors! I did not know about `CSSStyleDeclaration`; that's great.

I think I've addressed all the issues you pointed out. Let me know if there's anything else that should be done.

**Pull request merged**